vmhost/contexts/storage.go

```diff
@@ -425,3 +436,34 @@ func (context *storageContext) IsUseDifferentGasCostFlagSet() bool {
 func (context *storageContext) IsInterfaceNil() bool {
 	return context == nil
 }
+
+// GetStorageLoadCost returns the gas cost for the storage load operation
+func (context *storageContext) GetStorageLoadCost(trieDepth int64, staticGasCost uint64) (uint64, error) {
+	if context.host.EnableEpochsHandler().IsDynamicGasCostForDataTrieStorageLoadEnabled() {
+		return computeGasForStorageLoadBasedOnTrieDepth(trieDepth, context.host.Metering().GasSchedule().DynamicStorageLoad, staticGasCost)
+	}
+
+	return staticGasCost, nil
+}
+
+func computeGasForStorageLoadBasedOnTrieDepth(trieDepth int64, coefficients config.DynamicStorageLoadCostCoefficients, staticGasCost uint64) (uint64, error) {
+	fx := coefficients.Quadratic*trieDepth*trieDepth + coefficients.Linear*trieDepth + coefficients.Constant
```

Comment:
Please see https://github.com/multiversx/mx-chain-vm-go/blob/878997230e52736d9497dde8de9d4abdd0305cad/vmhost/contexts/storage.go#L535 for an overflow-safe implementation.

Reply:
Done.